4-projects/modules/infra_pipelines/main.tf

```diff
@@ -22,7 +22,7 @@ locals {
   state_buckets          = { for created_csr in local.created_csrs : "${created_csr}-tfstate" => format("%s-%s-%s", created_csr, "tfstate", random_id.suffix.hex) }
   apply_branches_regex   = "^(${join("|", var.terraform_apply_branches)})$"
   cloudbuild_bucket_name = "${var.cloudbuild_project_id}_cloudbuild"
-  cloudbuild_bucket      = { "cloudbuild" = local.cloudbuild_bucket_name }
+  cloudbuild_bucket      = { "${data.google_project.cloudbuild_project.project_id}_cloudbuild" = local.cloudbuild_bucket_name }
```

Comment:
probably the problem is that 
https://github.com/terraform-google-modules/terraform-example-foundation/blob/7a533bf0f41bb2e6681c4c58267eb2cce795206c/4-projects/modules/infra_pipelines/main.tf#L162-L165

the call to `gcloud builds submit` , is not waiting for the bucket to be created which would cause the collision.

It should also wait for `google_storage_bucket.pipeline_infra`

Reply:
https://cloud.google.com/build/docs/running-builds/start-build-command-line-api

>When you run gcloud builds submit for the first time in a Cloud project, Cloud Build creates a Cloud Storage bucket named [YOUR_PROJECT_NAME]_cloudbuild in that project. 
>Cloud Build uses this bucket to store any source code that you might use for your builds.

Reply:
> probably the problem is that
> 
> https://github.com/terraform-google-modules/terraform-example-foundation/blob/7a533bf0f41bb2e6681c4c58267eb2cce795206c/4-projects/modules/infra_pipelines/main.tf#L162-L165
> 
> the call to `gcloud builds submit` , is not waiting for the bucket to be created which would cause the collision.
> 
> It should also wait for `google_storage_bucket.pipeline_infra`

Thanks @daniel-cit and agreed.  The original fix plan was to add a random_id suffix as you suggest, however this was to confirm the expected `cloudbuild_project_id` value as keys are displayed during the TF plan, and [cloudbuild_project_id](https://github.com/terraform-google-modules/terraform-example-foundation/blob/7a533bf0f41bb2e6681c4c58267eb2cce795206c/4-projects/modules/single_project/main.tf#L28) should have a random suffix to prevent namespace collisions.

Good catch on the `gcloud builds submit` also creating the bucket! When I get a chance I'll investigate further unless you prefer to submit a PR.  Cheers!

Reply:
@apeabody PR https://github.com/terraform-google-modules/terraform-example-foundation/pull/719 create for this fix.

Thanks for rising the issue.

Reply:
Thanks @daniel-cit! Closing this in favor of your PR #719